### ISSUE TITLE
plot_state_city() cuts off x/y axes tick labels

### ISSUE THREAD

**Issue by derivation** [Qiskit/qiskit-terra#1869]
### Information

- **Qiskit Terra version**: 0.8
- **Python version**: 3.6.8
- **Operating system**: macOS 10.14.3

### What is the current behavior?
`plot_state_city()` cuts off x/y axes tick labels

`plot_state_city()` is defined [here](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/tools/visualization/_state_visualization.py)

### Steps to reproduce the problem

```
import numpy as np
from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister
from qiskit import execute
from qiskit.tools.visualization import plot_state_city
import matplotlib.pyplot as plt
q = QuantumRegister(3, 'q')
circ = QuantumCircuit(q)
circ.h(q[0])
circ.cx(q[0], q[1]).
circ.cx(q[0], q[2])
from qiskit import BasicAer
backend = BasicAer.get_backend('statevector_simulator')
job = execute(circ, backend)
result = job.result()
outputstate = result.get_statevector(circ)
plot_state_city(outputstate)

```

### What is the expected behavior?

The axes tick labels should not be cutoff.

### Suggested solutions

https://stackoverflow.com/questions/6774086/why-is-my-xlabel-cut-off-in-my-matplotlib-plot

**Comment by mtreinish** [comment]
Yeah I would just add a tight_layout() call to the end of the function. That's normally how I solve the cut off label problem

**Comment by mtreinish** [comment]
For reference the output with the labels cut off looks like this: ![test](https://user-images.githubusercontent.com/2447371/53518383-e99af080-3a9e-11e9-84e7-577c0919dd54.png)

**Comment by mtreinish** [comment]
Hmm, actually I just checked the code we have a `tight_layout()` call in there already: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/tools/visualization/_state_visualization.py#L339 so something else is going on. I'll have to dig more deeply.

**Comment by Knabberjoe** [comment]
I think this issue has been resolved with #3123. At least for me the ticks are not cut off anymore. (The suggested solution from the stackexchange website to simply add a small offset
`plt.subplots_adjust(bottom=0.15)`
worked in connection with the tight_layout() call as well.)

**Comment by 1ucian0** [comment]
solved via #3123. Feel free to reopen if that's not the case.
